### PR TITLE
A number of small tweaks to the MVP.

### PR DESCRIPTION
**Pull request by mccheah**
- Master protocol defaults to https if not specified
- Removed upload driver extra classpath functionality
- Added ability to specify main app resource with container:// URI
- Shut down the REST server and thus the driver pod once the application completes
- Updated docs to reflect all of the above
- Add examples to Docker images, mostly for integration testing but could be useful for easily getting started without shipping anything to the cluster

### CODE REVIEW

**Comment by mccheah** [comment]
@foxish @ash211 @erikerlandson bunch of small things here, though in retrospect each of these should be its own pull request.

File: docs/running-on-kubernetes.md
Comment (ash211) [inline docs/running-on-kubernetes.md]: Provide an example of what this looks like.  The double protocol thing is confusing as-is

Comment (mccheah) [reply docs/running-on-kubernetes.md]: Done.

File: resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/rest/kubernetes/KubernetesSparkRestServer.scala
Comment (foxish) [inline resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/rest/kubernetes/KubernetesSparkRestServer.scala:267]: Why was `barrier` added?

Comment (mccheah) [reply resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/rest/kubernetes/KubernetesSparkRestServer.scala:267]: The barrier needs to count down to unblock the main thread whenever we stop the server.

**Comment by ash211** [comment]
@mccheah can you please handle merge conflicts?

@foxish anything else you'd want to see changed here before merging?

**Comment by foxish** [comment]
I still haven't gone over this change completely as I do not completely understand some parts of it. I'd like to go over it once tonight. Can we merge this tomorrow?

**Comment by foxish** [comment]
LGTM

**Comment by mccheah** [comment]
@foxish @ash211 fixed merge conflicts. Integration tests were also failing from the pod and service name change, so fixed those also.

**Review by ash211 — APPROVED** [review]
@foxish @erikerlandson either of you have any concerns?  Otherwise let's merge this

**Comment by erikerlandson** [comment]
LGTM

**Pull request merged**
